### ISSUE TITLE
GS1 always starts with Character Set B

### ISSUE THREAD

**Issue by noperator-zz** [WhyNotHugo/python-barcode#110]
Some codes can be encoded entirely with code C, but Code128() initializes self._charset = "B" unconditionally.
Also, due to the (non-numeric) FNC 1 character at the start of a GS1 barcode, the library will always start with character set B anyway, even if self._charset was set to "C". https://github.com/WhyNotHugo/python-barcode/blob/83a0b4e87755f8dcb22c9cf35c52f6f3ae7a6993/barcode/codex.py#L175

The effect is that a switch to Code C is added after FNC 1, when it could have just started with character set C instead.

Expected (ignore the slash):
START C -> FNC 1 -> 12 -> 00 -> 13 -> 00 -> 85
![image](https://user-images.githubusercontent.com/12838439/123175242-4cfe7880-d436-11eb-9a30-13fa0162a0dc.png)

Actual:
START B -> FNC 1 -> CODE C -> 12 -> 00 -> 13 -> 00 -> 85
![gs128](https://user-images.githubusercontent.com/12838439/123175445-a9fa2e80-d436-11eb-872c-999169cb0348.png)

```
import barcode
from barcode.writer import ImageWriter
sn = 1200130085
# START B -> FNC 1 -> CODE C -> barcode (code C)
code = barcode.get("gs1_128", str(sn), writer=ImageWriter())
code.save("gs128")
```

**Comment by audoh-tickitto** [comment]
Similar issue with Code128, it starts with charset B so it will never switch to charset A as A is a subset of B.

**Comment by WhyNotHugo** [comment]
Yup, makes sense. The "next charset" should be evaluated at the rather than hardcoded.

**Comment by lhopkins** [comment]
@WhyNotHugo Any timeline for a fix on this? I'm happy to submit a PR.

**Comment by WhyNotHugo** [comment]
PRs welcome

**Comment by MrWangg1992** [comment]
any updates for this bug ?

**Comment by yohan-hicof** [comment]
Hello,
I encountered the same issue.
There is probably a simple fix for that.
First, since this is a GS1 code, we know that after the FNC1 we will have at least two digit for the AI.
So in the class Gs1_128 add this line at the end of the init:
`self._charset = "C"`
Then, in the Code128 class, function _maybe_switch_charset, replace:
`if self._charset == "C" and not char.isdigit():`
by
`if self._charset == "C" and not (char.isdigit() or ord(char)==241):`

I did not fully test the second change, but since the FNC1 char is in all three charset, changing the charset before or after the FNC1 should not have any impact on any code.
